src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4338,6 +4345,9 @@ private void DoConnect(EndPoint endPointSnapshot, Internals.SocketAddress socket
             // Throw an appropriate SocketException if the native call fails.
             if (errorCode != SocketError.Success)
             {
+                SocketsTelemetry.Log.ConnectFailed();
```

Comment:
It seems like every ConnectFailed is followed immediately by a ConnectStop.  Can't the former just do the latter and save the boilerplate/overhead at the call site?

Reply:
Calls merged.